### PR TITLE
Hotfix/reuse sublib

### PR DESCRIPTION
**Pull request by Omnielement**
reuse sublib

### CODE REVIEW

**Review by Setnour6 — CHANGES_REQUESTED** [review]
The firefly pylon does not appear, the MothWorldDoor tile and FireflyCapital item does not work, and the Yggdrasil Kelp Background music does not play. There is also an issue where after a world is finished being created, a black screen appears with the text "Validating World Save: 100%" and you get stuck. See issue #179 for more information.

They need to be fixed.

**Comment by Omnielement** [comment]
> The firefly pylon does not appear, the MothWorldDoor tile and FireflyCapital item does not work, and the Yggdrasil Kelp Background music does not play. There is also an issue where after a world is finished being created, a black screen appears with the text "Validating World Save: 100%" and you get stuck. See issue #179 for more information.
> 
> They need to be fixed.

Something was needed in the firefly tile hotfix,and I'll fix the rest part of these bug soon.

**Comment by Omnielement** [comment]
> The firefly pylon does not appear, the MothWorldDoor tile and FireflyCapital item does not work, and the Yggdrasil Kelp Background music does not play. There is also an issue where after a world is finished being created, a black screen appears with the text "Validating World Save: 100%" and you get stuck. See issue #179 for more information.
> 
> They need to be fixed.

fixed

File: Sources/Modules/CagedDomain/Everglow.CagedDomain.csproj [hunk 1]
```diff
@@ -1,5 +1,10 @@
 
     <Project Sdk="Microsoft.NET.Sdk">
       <Import Project="..\Module.props" />
+	<ItemGroup>
+	  <Reference Include="SubworldLibrary">
+	    <HintPath>..\..\..\Libraries\SubworldLibrary.dll</HintPath>
+	  </Reference>
+	</ItemGroup>
```
Comment (Setnour6) [inline Sources/Modules/CagedDomain/Everglow.CagedDomain.csproj:8]: why does CagedDomain reference Subworld Library? Are you planning on adding subworld functions to that module?

Comment (Omnielement) [reply Sources/Modules/CagedDomain/Everglow.CagedDomain.csproj:8]: yes，this branch maybe work with a subWorld.

**Review by Setnour6 — CHANGES_REQUESTED** [review]
Upon testing, I revealed the entire map, teleported to the moth world door, and entered it. There was no moth underground background showing
image

Only for a brief second does the background show before it becomes black completely
image

The entire world resets when I exit the firefly subworld through the moth world door.
image

That does not happen when I click on the return button on the settings menu. It also temporarily fixes the world-resetting bug, but not for the subworld.

So what I request is that
1. The moth underground background can fully appear again
2. The subworld can be saved again
3. The main world does not reset after exiting the subworld through the moth world door (or firefly capital/impression item).

I will test the Yggdrasil subworld and either edit this comment post or make a new one

**Comment by Omnielement** [comment]
> 3\. orld through the moth world door (or firefly capital/impression item).
> 
> ### Yggdrasil Subworld

The field "ShouldSave" in the SubWorld class, if true, will lead to crash when creating new world.(Sublib should be blame in this case).And, if false, all block changes will not save when leaving the subWorld.

**Review by Setnour6 — APPROVED** [review]
Although the subworld cannot be saved, all of the other issues have been fixed. The subworld saving fix can be re-implemented after the saving-crash issue has been fixed.
There is still an issue where the main world is reset after exiting the subworld, but that might also be fixed in a future PR or sublib fix.
I approve of this PR

**Pull request merged**